exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2419 @@
+
+func TestFuturesEndFundingFeeSettlement(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresEndFundingFeeSettlementPushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresStartFundingFeeSettlementPushDataJSON = `{ "topic": "/contract/announcement","subject": "funding.begin","data": {"symbol": "XBTUSDM","fundingTime": 1551770400000,"fundingRate": -0.002966,"timestamp": 1551770400000}}`
+
+func TestFuturesStartFundingFeeSettlementPushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresStartFundingFeeSettlementPushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresFundingRatePushDataJSON = `{ "topic": "/contract/instrument:XBTUSDM","subject": "funding.rate","data": {"granularity": 60000,"fundingRate": -0.002966,"timestamp": 1551770400000}}`
+
+func TestFuturesFundingRatePushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresFundingRatePushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresMarkIndexPricePushDataJSON = `{ "topic": "/contract/instrument:XBTUSDM","subject": "mark.index.price","data": {"granularity": 1000,"indexPrice": 4000.23,"markPrice": 4010.52,"timestamp": 1551770400000}}`
+
+func TestFuturesMarkIndexPricePushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresMarkIndexPricePushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGenerateDefaultSubscriptions(t *testing.T) {
+	t.Parallel()
+	if _, err := ku.GenerateDefaultSubscriptions(); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetAvailableTransferChains(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	if _, err := ku.GetAvailableTransferChains(context.Background(), currency.BTC); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	if _, err := ku.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Futures); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", ku.Name, err)
+	}
+}
+
+func TestGetOrderInfo(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("Kucoin GetOrderInfo() skipping test: api keys not set")
+	}
+	enabled, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Error("couldn't find enabled tradable pairs")
+	}
+	if len(enabled) == 0 {
+		t.Skip(cantManipulateRealOrdersOrKeysNotSet)
+	}
+	_, err = ku.GetOrderInfo(context.Background(), "123", enabled[0], asset.Futures)
+	if err != nil && !strings.Contains(err.Error(), "orderNotExist") {
+		t.Errorf("Kucoin GetOrderInfo() expecting %s, but found %v", "Order does not exist", err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	if _, err := ku.GetDepositAddress(context.Background(), currency.BTC, "", ""); err != nil && errors.Is(err, errNoDepositAddress) {
```

Comment:
I think you mean this:
```suggestion
	if _, err := ku.GetDepositAddress(context.Background(), currency.BTC, "", ""); err != nil && !errors.Is(err, errNoDepositAddress) {
```

Reply:
Yes, it is!